### PR TITLE
🧹semantic tests & 🐙separate packages w/ lerna

### PR DESCRIPTION
**Pull request by jperl**
Resolves https://github.com/qawolf/qawolf/issues/72
Resolves https://github.com/qawolf/qawolf/issues/61

### CODE REVIEW

**Comment by jperl** [comment]
Need to import the @qawolf/jest-environment in setupFilesAfterEnv if you want the global types to work. For the tests in @qawolf/jest-environment we do it with setup.ts.

https://github.com/qawolf/qawolf/blob/48ae51cfa21e303c7536c507ac391636fa6a9793/packages/jest-environment/setup.ts

https://github.com/qawolf/qawolf/blob/48ae51cfa21e303c7536c507ac391636fa6a9793/packages/jest-environment/jest.config.js#L9

**Comment by jperl** [comment]
Going to setup e2e tests in a separate PR

File: packages/browser/src/Browser.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
-import puppeteer, { Page } from "puppeteer";
+import { logger } from "@qawolf/logger";
+import { CONFIG } from "@qawolf/config";
```
Comment (flaurida) [inline packages/browser/src/Browser.ts]: ABC

File: packages/build-test/static/test.hbs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+describe('{{{name}}}', () => {
```
Comment (flaurida) [inline packages/build-test/static/test.hbs]: i'd like the test to at least include info about what it is entering into inputs

File: packages/build-test/tests/__snapshots__/buildTest.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+"describe('login', () => {
+  it('can scroll down', async () => {
+    await scroll(steps[0]};
+  });
+
+  it('can click form authentication link', async () => {
+    await click(steps[1]);
+  });
+
+  it('can input into username input', async () => {
```
[the remark this answers is not in the document]
Comment (jperl) [reply packages/build-test/tests/__snapshots__/buildTest.test.ts.snap]: I think it's simpler not too that way we do not need to worry about secrets.

File: packages/cli/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    await runner.close();
+
+    process.exit(0);
+  });
+
+program
+  .command("test [name]")
+  .description("run a test")
+  .action(async name => {
+    const success = await runTest(name ? snakeCase(name) : undefined);
```
Comment (flaurida) [inline packages/cli/src/index.ts]: i prefer `null` over `undefined`

File: packages/jest-environment/src/RunnerEnvironment.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { logger } from "@qawolf/logger";
+import { Runner } from "@qawolf/runner";
+import { Job } from "@qawolf/types";
+import { EnvironmentContext } from "@jest/environment";
```
Comment (flaurida) [inline packages/jest-environment/src/RunnerEnvironment.ts]: abc

**Pull request merged**